replication/client_lib/src/rrdb_client_factory_impl.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+#include <dsn/dist/replication/replication_app_client_base.h>
+#include "rrdb_client.h"
+#include "rrdb_error.h"
+#include "rrdb_client_impl.h"
+
+using namespace dsn::replication;
+
+extern void dsn_core_init();
+
+namespace dsn{ namespace apps{
+
+class rrdb_client_factory_impl{
+public:
+    static bool initialize(const char* config_file)
+    {
+        dsn_core_init();
+        //use config file to run
+        char exe[] = "client";
+        char config[50];
+        sprintf(config, "%s", config_file);
+        char* argv[] = { exe, config};
+        dsn_run(2, argv, false);
+        rrdb_client_impl::init_error();
+        return true;
+    }
+
+    static irrdb_client* create_client(const char* app_name, const char* cluster_name)
```

Comment:
I thought the factory is used for avoiding creating multiple clients for the same app-bame+cluster. If it is purely for creating the client, why not just client::create if you want to preserve Irrdb-client? Or even simpler like new client if you don't need to keep IrDA-client.